src/lib/mongoBackend/dateExpiration.cpp

```diff
@@ -0,0 +1,200 @@
+/*
+*
+* Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Fermin Galan
+*
+*/
+#include <string>
+#include <vector>
+
+#include "common/string.h"
+#include "common/globals.h"
+#include "logMsg/logMsg.h"
+#include "ngsi/ContextAttribute.h"
+#include "parse/CompoundValueNode.h"
+#include "rest/OrionError.h"
+
+// FIXME P5: the following could be not necessary if we optimize the valueBson() thing. See
+// the next FIXME P5 comment in this file
+#include "mongoBackend/safeMongo.h"
+#include "mongoBackend/dbConstants.h"
+#include "mongoBackend/dateExpiration.h"
+
+
+
+/* ****************************************************************************
+*
+* USING
+*/
+using mongo::BSONObjBuilder;
+using mongo::BSONArrayBuilder;
+using orion::CompoundValueNode;
+
+
+
+/* ****************************************************************************
+*
+* getDateExpiration -
+*
+* Get the ISO8601 Expiration Date information for the given
+* ContextAttribute provided as parameter.
+*
+* It returns true, except in the case of error (in which in addition errDetail gets filled)
+*/
+static bool getDateExpiration
+(
+  const ContextAttribute*  caP,
+  mongo::Date_t*           dateExpiration,
+  std::string*             errDetail
+)
+{
+  if ((caP->type == DATE_TYPE) || (caP->type == DATE_TYPE_ALT))
+  {
+    *dateExpiration = mongo::Date_t(caP->numberValue*1000);
```

Comment:
Why *1000 ?

Reply:
Because the value transformed by the upper parsing method (the one that fills numberValue starting  from the ISO8601 string in input), seems to be filled with seconds, when we create the mongo::Date_t object, we want milliseconds in the related constructor.

Reply:
Ok. Clear. NTC.